### DISCUSSION TITLE
How to use react hook useState with zustand in nextJS

### DISCUSSION THREAD

**Discussion by marvthedev** [pmndrs/zustand#765] [Q&A]
How can I pass useState values to the store? I can't use useState since initStore() is not a functional component. I was trying to [use this example.](https://codesandbox.io/s/nextjs-with-zustand-forked-rt7yt?file=/pages/index.js)

my store.js
`import { useMemo } from "react";
import create from "zustand";

let store;

function initStore() {
  return create((set) => ({
    isCartOpen: false,
    setCartOpen: () => set((state) => ({ isCartOpen: !state.isCartOpen })),
  }));
}

export const initializeStore = (preloadedState) => {
  let _store = store ?? initStore(preloadedState);

  // After navigating to a page with an initial Zustand state, merge that state
  // with the current state in the store, and create a new store
  if (preloadedState && store) {
    _store = initStore({
      ...store.getState(),
      ...preloadedState,
    });
    // Reset the current store
    store = undefined;
  }

  // For SSG and SSR always create a new store
  if (typeof window === "undefined") return _store;
  // Create the store once in the client
  if (!store) store = _store;

  return _store;
};

export function useHydrate(initialState) {
  const state =
    typeof initialState === "string" ? JSON.parse(initialState) : initialState;
  const store = useMemo(() => initializeStore(state), [state]);
  return store;
}
`

**Comment by dai-shi** [comment]
Hi, have you looked into `zustand/context`?

**Reply by marvthedev** [reply]
I see that. But how would I re-work all this code to use zustand/context?

**Reply by dai-shi** [reply]
> How can I pass useState values to the store?

I might have misunderstood the question. So, you want to update the zustand store based on a value from useState? Then, `useEffect` is a solution.

**Reply by marvthedev** [reply]
Sorry dont mean to be confusing. I want to create a useState (example: const [ isCartOpen, setCartOpen ] = useState(false)). I want to use isCartOpen and setCartOpen as values for items inside my store. For instance, cartOpen: isCartOpen, setCartOpen: setCartOpen

**Reply by dai-shi** [reply]
You can store those values in your store. It might not be a good design though, mixing local state and global state outside React. I would put isCartOpen state in the store from the beginning.
